core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -79,6 +81,7 @@ private[spark] class ApplicationInfo(
     val exec = new ExecutorDesc(newExecutorId(useID), this, worker, cores, desc.memoryPerSlave)
     executors(exec.id) = exec
     coresGranted += cores
+    coresMax = if(coresGranted > coresMax) coresGranted else coresMax
```

Comment:
Might be able to use `math.max` here. Also should the column heading say "Cores (max)" in the case of completed apps only? Otherwise LGTM.